Userland/Libraries/LibJS/Runtime/FinalizationRegistry.cpp

```diff
@@ -54,30 +50,34 @@ void FinalizationRegistry::remove_dead_cells(Badge<Heap>)
         break;
     }
     if (any_cells_were_removed)
-        vm().enqueue_finalization_registry_cleanup_job(*this);
+        vm().host_enqueue_finalization_registry_cleanup_job(*this);
 }
 
 // 9.13 CleanupFinalizationRegistry ( finalizationRegistry ), https://tc39.es/ecma262/#sec-cleanup-finalization-registry
-ThrowCompletionOr<void> FinalizationRegistry::cleanup(FunctionObject* callback)
+ThrowCompletionOr<void> FinalizationRegistry::cleanup(Optional<JobCallback> callback)
 {
+    auto& vm = this->vm();
+    auto& global_object = this->global_object();
+
     // 1. Assert: finalizationRegistry has [[Cells]] and [[CleanupCallback]] internal slots.
     // Note: Ensured by type.
 
     // 2. Let callback be finalizationRegistry.[[CleanupCallback]].
-    auto cleanup_callback = callback ?: m_cleanup_callback;
+    auto& cleanup_callback = callback.has_value() ? callback.value() : m_cleanup_callback;
 
     // 3. While finalizationRegistry.[[Cells]] contains a Record cell such that cell.[[WeakRefTarget]] is empty, an implementation may perform the following steps:
     for (auto it = m_records.begin(); it != m_records.end(); ++it) {
         // a. Choose any such cell.
         if (it->target != nullptr)
             continue;
-        auto cell = *it;
 
         // b. Remove cell from finalizationRegistry.[[Cells]].
+        MarkedValueList arguments(vm.heap());
+        arguments.append(it->held_value);
         it.remove(m_records);
 
         // c. Perform ? HostCallJobCallback(callback, undefined, « cell.[[HeldValue]] »).
-        (void)TRY(call(global_object(), *cleanup_callback, js_undefined(), cell.held_value));
+        (void)vm.host_call_job_callback(global_object, cleanup_callback, js_undefined(), move(arguments));
```

Comment:
Is the removed `TRY` intentional here? It does say `Perform _?_ Hos...`

Reply:
That's a mistake! Great catch :)